lib/hackerSortingAlgo.ts

```diff
@@ -0,0 +1,103 @@
+const fs = require('fs')
+const path = require('path')
+const { parse } = require('csv-parse/sync')
+
+// parse csv file into an array
+function loadCSV (filepath: string): any[] {
+  const csvFileAbsolutePath = path.resolve(__dirname, 'data', filepath)
+
+  // error handling in case file is missing
+  let fileContent
+  try {
+    fileContent = fs.readFileSync(csvFileAbsolutePath, {
+      encoding: 'utf-8'
+    })
+  } catch (err) {
+    console.log(`File cannot be found : "${filepath}"`)
+    return []
+  }
+
+  const options = {
+    delimiter: ',',
+    columns: true
+  }
+  return parse(fileContent, options)
+}
+
+// loops through each user row in the given array
+// -> for each question: increment count for corresponding cabin if answers match
+function matchAnswers (hackerList: any[], answerList: any[], cabinList: any[]) {
```

Comment:
The `matchAnswers` function is quite long and contains nested loops, which can make the code harder to understand and maintain. Consider breaking it down into smaller, more focused functions to improve readability and maintainability. I think breaking it into two or three functions would do the trick. Remember from OOD—**(S)**INGLE RESPONSIBILITY **O.L.I.D**